src/lib.rs

```diff
@@ -0,0 +1,886 @@
+
+const CHANNEL_BUF_SIZE: usize = 1000;
+
+#[derive(Debug, Clone)]
+/// Represents the configuration of an `LdkLite` instance.
+pub struct LdkLiteConfig {
+	/// The path where the underlying LDK and BDK persist their data.
+	pub storage_dir_path: String,
+	/// The URL of the utilized Esplora server.
+	pub esplora_server_url: String,
+	/// The used Bitcoin network.
+	pub network: bitcoin::Network,
+	/// The TCP port the network node will listen on.
+	pub listening_port: u16,
+	/// The default CLTV expiry delta to be used for payments.
+	pub default_cltv_expiry_delta: u32,
+}
+
+/// A builder for an [`LdkLite`] instance, allowing to set some configuration and module choices from
+/// the getgo.
+#[derive(Debug, Clone)]
+pub struct LdkLiteBuilder {
+	config: LdkLiteConfig,
+}
+
+impl LdkLiteBuilder {
+	/// Creates a new builder instance with the default configuration.
+	pub fn new() -> Self {
+		// Set the config defaults
+		let storage_dir_path = "/tmp/ldk_lite/".to_string();
+		let esplora_server_url = "https://blockstream.info/api".to_string();
+		let network = bitcoin::Network::Testnet;
+		let listening_port = 9735;
+		let default_cltv_expiry_delta = 144;
+
+		let config = LdkLiteConfig {
+			storage_dir_path,
+			esplora_server_url,
+			network,
+			listening_port,
+			default_cltv_expiry_delta,
+		};
+
+		Self { config }
+	}
+
+	/// Creates a new builder instance from an [`LdkLiteConfig`].
+	pub fn from_config(config: LdkLiteConfig) -> Self {
+		Self { config }
+	}
+
+	/// Sets the used storage directory path.
+	///
+	/// Default: `/tmp/ldk_lite/`
+	pub fn storage_dir_path(&mut self, storage_dir_path: String) -> &mut Self {
+		self.config.storage_dir_path = storage_dir_path;
+		self
+	}
+
+	/// Sets the Esplora server URL.
+	///
+	/// Default: `https://blockstream.info/api`
+	pub fn esplora_server_url(&mut self, esplora_server_url: String) -> &mut Self {
+		self.config.esplora_server_url = esplora_server_url;
+		self
+	}
+
+	/// Sets the Bitcoin network used.
+	///
+	/// Options: `mainnet`/`bitcoin`, `testnet`, `regtest`, `signet`
+	///
+	/// Default: `testnet`
+	pub fn network(&mut self, network: &str) -> &mut Self {
+		self.config.network = match network {
+			"mainnet" => bitcoin::Network::Bitcoin,
+			"bitcoin" => bitcoin::Network::Bitcoin,
+			"testnet" => bitcoin::Network::Testnet,
+			"regtest" => bitcoin::Network::Regtest,
+			"signet" => bitcoin::Network::Signet,
+			_ => bitcoin::Network::Testnet,
+		};
+		self
+	}
+
+	/// Sets the port on which [`LdkLite`] will listen for incoming network connections.
+	///
+	/// Default: `9735`
+	pub fn listening_port(&mut self, listening_port: u16) -> &mut Self {
+		self.config.listening_port = listening_port;
+		self
+	}
+
+	/// Builds an [`LdkLite`] instance according to the options previously configured.
+	pub fn build(self) -> Result<LdkLite, Error> {
+		let config = Arc::new(self.config);
+
+		let ldk_data_dir = format!("{}/ldk", &config.storage_dir_path.clone());
```

Comment:
Yeah, I was debating this, since `ldk_data_dir`/`bdk_data_dir` are really dependent variables on `storage_dir_path`. Do you think we should give up a singular `storage_dir_path` and instead let the user configure the individual data directories entirely separately?